### PR TITLE
Fix: Attachment endpoint to save file against automation rule

### LINKED ISSUE #4356: Add support for new automation actions
**Missing Actions:**

- Send Message
- Send Attachment
- Send email to team

**Missing Conditions:**

- Add conditions based on contact and conversations for all the events

@tejaswinichile To provide a sample payload, Postman collection, description about attributes.
@fayazara to work on a custom component to implement the actions.

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Added attachment endpoint to upload the file then send the blob inside the action params as send attachment.

Fixes #4356 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Added the test cases and tested on local.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
https://www.loom.com/share/ee42bdc4040f4a9e95423532f4d54ac3

Loom video about, how the new attachment endpoint would work in the flow of saving automation with attachments.
cc @fayazara Let me know if you have any doubt.

**Comment by tejaswinichile** [comment]
rule_file_json

@fayazara Check the files JSON in the automation rule JSON.

**Comment by tejaswinichile** [comment]
https://www.loom.com/share/0e670a94d800473c89e4877f5151d18d

@pranavrajs @fayazara Please have a look at this one.

File: spec/controllers/api/v1/accounts/automation_rules_controller_spec.rb [hunk 1]
```diff
@@ -133,60 +133,73 @@
...
+
+        expect(blob_2['blob_id']).to be_present
+
         params[:actions] = [
           {
-            action_name: :send_message,
-            action_params: ['Welcome to the chatwoot platform.']
-          },
-          {
-            action_name: :update_additional_attributes,
```
Comment (tejaswinichile) [inline spec/controllers/api/v1/accounts/automation_rules_controller_spec.rb:174]: removed other conditions to resolve the RuboCop issues, we do have these conditions in other test cases.

**Comment by pranavrajs** [comment]
@tejaswinichile Can you please check this error? https://sentry.io/share/issue/933a2f268e674afaa29c96a56acab825/

File: app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue [hunk 2]
```diff
@@ -507,6 +513,15 @@ export default {
       if (type === null) return false;
       return true;
     },
+    getFileName(id, actionType) {
+      if (!id) return '';
+      if (actionType === 'send_attachment') {
+        const file = this.automation.files.find(item => item.blob_id === id);
+        // replace `blob_id.toString()` with file name once api is fixed.
+        if (file) return file.blob_id.toString();
```
Comment (fayazara) [inline app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue:521]: @tejaswinichile Once file name is added to the files `array`, can you please replace this `blob_id.toString()` with the file name key. Added this temporarily.

File: app/models/automation_rule.rb [hunk 3]
```diff
@@ -28,7 +30,21 @@ class AutomationRule < ApplicationRecord
...
+  def file_base_data
+    files.map do |file|
+      {
+        id: file.id,
+        automation_rule_id: id,
+        file_type: file.content_type,
+        account_id: account_id,
+        file_url: url_for(file),
+        blob_id: file.blob_id,
+        thumb_url: url_for(file.representation(resize: '250x250'))
```
Comment (pranavrajs) [inline app/models/automation_rule.rb:44]: Remove this?

File: app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue (hunk 2 above)
Comment (tejaswinichile) [reply app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue]: multiple file attachment

@fayazara, I have resolved the filename issue and multiple file uploads not showing up case.

**Comment by tejaswinichile** [comment]
@sojan-official This is ready for review, can you check backend code.

**Review by sojan-official — APPROVED** [review]
ruby code LGTM

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**